File: app/src/main/java/com/originb/inkwisenote2/InkWiseApplication.java
package com.originb.inkwisenote2;

import android.app.Application;
import androidx.work.Configuration;
import androidx.work.WorkManager;

public class InkWiseApplication extends Application {

    @Override
    public void onCreate() {
        super.onCreate();

        // Initialize WorkManager with Koin factory before any other initialization
        // This prevents the "WorkManager is already initialized" error
        Configuration workManagerConfiguration = new Configuration.Builder()
                .setWorkerFactory(KoinWorkManagerHelperKt.getKoinWorkManagerFactory())
                .build();
        WorkManager.initialize(this, workManagerConfiguration);
    }
}

File: app/src/main/java/com/originb/inkwisenote2/InkWiseApplication.kt
package com.originb.inkwisenote2

import android.app.Application
import androidx.work.Configuration
import androidx.work.WorkManager

class InkWiseApplication : Application() {
    override fun onCreate() {
        super.onCreate()

        // Initialize WorkManager with Koin factory before any other initialization
        // This prevents the "WorkManager is already initialized" error
        val workManagerConfiguration = Configuration.Builder()
            .setWorkerFactory(getKoinWorkManagerFactory())
            .build()
        WorkManager.initialize(this, workManagerConfiguration)
    }
}
